python/packages/ai/teams/ai/promptsv2/prompt_functions.py

```diff
@@ -0,0 +1,79 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any, Callable, List
+
+from botbuilder.core import TurnContext
+
+from ...state import Memory
+from ..tokenizers import Tokenizer
+
+"""
+A function that can be called from a prompt template string.
+
+Parameters:
+    context (TurnContext): Context for the current turn of conversation.
+    memory (Memory): Interface used to access state variables.
+    functions (PromptFunctions): Collection of functions that can be called from a prompt template string.
+    tokenizer (Tokenizer): Tokenizer used to encode/decode strings.
+    args (List[str]): Arguments to the function as an array of strings.
+
+Returns:
+    Any: The return type is not specified, so it can be any type.
+"""
+PromptFunction = Callable[[TurnContext, Memory, "PromptFunctions", Tokenizer, List[str]], Any]
```

Comment:
should this return type be `Awaitable[Any]` for async/await?

Reply:
Thanks, will update with my next commit